test/integration/extension_discovery_integration_test.cc

```diff
@@ -775,5 +856,68 @@ TEST_P(ExtensionDiscoveryIntegrationTest, ReloadBoth) {
   codec_client_->close();
 }
 
+TEST_P(ExtensionDiscoveryIntegrationTest, BasicSuccessConfigDump) {
+  two_ecds_filters_ = true;
+  on_server_init_function_ = [&]() { waitXdsStream(); };
+  addDynamicFilter("foo", false);
+  addDynamicListenerFilter("bar");
+  initialize();
+  test_server_->waitForCounterGe("listener_manager.lds.update_success", 1);
+  EXPECT_EQ(test_server_->server().initManager().state(), Init::Manager::State::Initializing);
+  registerTestServerPorts({"http"});
+  sendXdsResponse("foo", "1", denyPrivateConfig());
+  sendXdsResponse2("bar", "2", 7);
+  test_server_->waitForCounterGe("extension_config_discovery.http_filter.foo.config_reload", 1);
+  test_server_->waitUntilListenersReady();
+  test_server_->waitForGaugeGe("listener_manager.workers_started", 1);
+  EXPECT_EQ(test_server_->server().initManager().state(), Init::Manager::State::Initialized);
+  BufferingStreamDecoderPtr response;
+  EXPECT_EQ("200", request("admin", "GET", "/config_dump", response));
+  EXPECT_EQ("application/json", contentType(response));
+  Json::ObjectSharedPtr json = Json::Factory::loadFromString(response->body());
+  size_t index = 0;
+  const std::string expected_types[] = {"type.googleapis.com/envoy.admin.v3.BootstrapConfigDump",
+                                        "type.googleapis.com/envoy.admin.v3.ClustersConfigDump",
+                                        "type.googleapis.com/envoy.admin.v3.EcdsConfigDump", // HTTP
```

Comment:
If both HTTP ECDS filter and listener ECDS filters are configured, then there will be two EcdsConfigDump entries in the /config_dump.  Since the data are stored in two different objects, there is no way to dump them into one EcdsConfigDump proto.
1) For listener ECDS filter: https://github.com/envoyproxy/envoy/blob/b0d79665240cdfe3d87280283d28ac30334542e9/source/server/listener_manager_impl.h#L111

2) For HTTP ECDS filter:
https://github.com/envoyproxy/envoy/blob/b0d79665240cdfe3d87280283d28ac30334542e9/source/extensions/filters/network/http_connection_manager/config.cc#L224